CDCgov/prime-reportstream
# refactor: move value sets gets to react-query

this is the first PR for creation of an MVP of react query based usage in the app.

this change covers:

* installation of react-query, creation of QueryClient Provider in app
* replacement of existing value sets data fetching hooks with react-query based hooks
* react query test tooling
* tests around new hooks and some additions and updates to tests for value sets index, which consumes them

## Test Steps:

Regression testing for value sets:

1. start servers on this branch
2. visit http://localhost:3000/admin/value-sets, logging in as necessary
3. _VERIFY_: value sets page loads as normal, see screenshot
4. click on `residence_type` value set
5. _VERIFY_: value sets detail page loads as normal, see screenshot
6. update or create a new a value set row, and save
5. _VERIFY_: your change is reflected on the page
6. refresh
5. _VERIFY_: value sets detail page loads correctly and includes your change

## Screenshots

<img width="1382" alt="Screen Shot 2022-08-23 at 5 22 45 PM" src="https://user-images.githubusercontent.com/92806979/186268775-141d444c-548e-4f77-a10e-938f1e97cb1a.png">

<img width="1361" alt="Screen Shot 2022-08-23 at 5 20 19 PM" src="https://user-images.githubusercontent.com/92806979/186268783-d639f216-56d0-4108-a8ae-508ed208fd1c.png">



## Checklist

### Testing
